database_manager/src/cli.rs

```diff
@@ -0,0 +1,227 @@
+pub use clap::{Arg, ArgAction, Args, Command, FromArgMatches, Parser};
+use clap_utils::get_color_style;
+use clap_utils::FLAG_HEADER;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Parser, Clone, Deserialize, Serialize, Debug)]
+#[clap(
+    name = "database_manager",
+    visible_alias = "db",
+    about = "Manage a beacon node database.",
+    styles = get_color_style(),
+    next_line_help = true,
+    term_width = 80,
+    disable_help_flag = true,
+    disable_help_subcommand = true,
+    display_order = 0,
+)]
+pub struct DatabaseManager {
+    #[clap(
+        long,
+        value_name = "SLOT_COUNT",
+        help = "Specifies how often a freezer DB restore point should be stored. \
+                Cannot be changed after initialization. \
+                [default: 2048 (mainnet) or 64 (minimal)]",
+        display_order = 0
+    )]
+    pub slots_per_restore_point: Option<u64>,
+
+    #[clap(
+        long,
+        value_name = "DIR",
+        help = "Data directory for the freezer database.",
+        display_order = 0
+    )]
+    pub freezer_dir: Option<PathBuf>,
+
+    #[clap(
+        long,
+        value_name = "EPOCHS",
+        default_value_t = 0,
+        help = "The margin for blob pruning in epochs. The oldest blobs are pruned \
+                up until data_availability_boundary - blob_prune_margin_epochs.",
+        display_order = 0
+    )]
+    pub blob_prune_margin_epochs: u64,
+
+    #[clap(
+        long,
+        value_name = "DIR",
+        help = "Data directory for the blobs database.",
+        display_order = 0
+    )]
+    pub blobs_dir: Option<PathBuf>,
+
+    #[clap(
+        long,
+        global = true,
+        help = "Prints help information",
+        action = clap::ArgAction::HelpLong,
+        display_order = 0,
+        help_heading = FLAG_HEADER
+    )]
+    help: Option<bool>,
+
+    #[clap(subcommand)]
+    pub subcommand: DatabaseManagerSubcommand,
+}
+
+#[derive(Parser, Clone, Deserialize, Serialize, Debug)]
+#[clap(rename_all = "kebab-case")]
+pub enum DatabaseManagerSubcommand {
+    Migrate(Migrate),
+    Inspect(Inspect),
+    Version(Version),
+    PrunePayloads(PrunePayloads),
+    PruneBlobs(PruneBlobs),
+    PruneStates(PruneStates),
+    Compact(Compact),
+}
+
+#[derive(Parser, Clone, Deserialize, Serialize, Debug)]
+#[clap(about = "Migrate the database to a specific schema version.")]
+pub struct Migrate {
+    #[clap(
+        long,
+        value_name = "VERSION",
+        help = "Schema version to migrate to",
+        display_order = 0
+    )]
+    pub to: u64,
+}
+
+#[derive(Parser, Clone, Deserialize, Serialize, Debug)]
+#[clap(about = "Inspect raw database values.")]
+pub struct Inspect {
+    #[clap(
+        long,
+        value_name = "TAG",
+        help = "3-byte column ID (see `DBColumn`)",
+        display_order = 0
+    )]
+    pub column: String,
+
+    // TODO InspectTarget::VARIANTS
```

Comment:
should we solve this TODO before merging?

Reply:
resolved in a32afe8